Src/Couchbase.Linq/N1QlFunctions.Core.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Couchbase.Linq
+{
+    /// <summary>
+    /// Implements static helper methods for N1QL queries
+    /// </summary>
+    public static partial class N1QlFunctions
+    {
+        /// <summary>
+        /// Shortcut for creating an error for methods that are only supported in N1QL, not in .Net.
+        /// </summary>
+        private static Exception NotImplementedError()
+        {
+            return new NotImplementedException("This method may only be used for generating N1QL queries.");
```

Comment:
@jeffrymorris 

Maybe we need to make it clearer.  Any calls to it are converted to N1QL by the query parsing process.  But you can't call it directly in .Net, only use it in lambda expressions that are never called.

Looking at the way EF does it, maybe it should be a NotSupportedException instead?

http://entityframework.codeplex.com/SourceControl/latest#src/EntityFramework.SqlServer/SqlFunctions.cs

Brant